### PR TITLE
Xen 4.12rc migration

### PR DESCRIPTION
**Pull request by aanisov**
Please note, here we have still adopted xen_git.bbappend from meta-xt-images. Switching to .inc files should be done as a separate task.

### CODE REVIEW

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend [hunk 1]
```diff
@@ -4,11 +4,11 @@ FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
 # Renesas R-Car
 ################################################################################
 
-XEN_REL_rcar = "4.10"
+XEN_REL_rcar = "4.12"
 PV = "${XEN_REL}.0+git${SRCPV}"
 SRCREV_rcar = "${AUTOREV}"
 
-SRC_URI_rcar = "git://github.com/xen-troops/xen.git;protocol=https;branch=master"
+SRC_URI_rcar = "git://github.com/xen-troops/xen.git;protocol=https;branch=xen-4.12rc-migration"
```
Comment (andr2000) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend]: I would rather think of moving current master, which is 4.10, to some tag/branch making sure all other dependent products use that. After that moving master to 4.12

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend [hunk 2]
```diff
@@ -25,7 +25,7 @@ SRC_URI_append_r8a7796 = " \
 # Generic
 ################################################################################
 
-FLASK_POLICY_FILE = "xenpolicy-${XEN_REL}.0"
+FLASK_POLICY_FILE = "xenpolicy-${XEN_REL}.0-rc"
```
Comment (andr2000) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend]: We currently have:
lrwxrwxrwx 2 xtfs xtfs        16 Jan 17 23:01 xenpolicy -> xenpolicy-4.10.0
so, just curios if this symlink be created properly

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/xen-chosen.dtsi [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
 / {
 	chosen {
-		bootargs = "dom0_mem=256M console=dtuart dtuart=serial0 dom0_max_vcpus=4 bootscrub=0 loglvl=all cpufreq=none sched=credit2 sched_ratelimit_us=100";
+		bootargs = "dom0_mem=256M console=dtuart dtuart=serial0 dom0_max_vcpus=4 bootscrub=0 loglvl=all cpufreq=none sched=credit2 sched_ratelimit_us=100 hmp-unsafe=true";
```
Comment (otyshchenko1) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/xen-chosen.dtsi]: Xen migration patch series doesn't contain CPUFreq stuff, so "cpufreq=none" is not needed I think

Comment (aanisov) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/xen-chosen.dtsi]: Right you are! I looked at it but left it occasionally.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend (hunk 2 above)
Comment (aanisov) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend]: Yes, the link is correct. On rc builds xenpolicy file has rc in its name.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend (hunk 1 above)
Comment (aanisov) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend]: xen-4.12rc-migration is a current feature branch. Actually it's a mock for now.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend (hunk 2 above)
Comment (andr2000) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend]: My point is that some of us might have their environment expecting the symlink without "rc" suffix...

**Comment by otyshchenko1** [comment]
With comments being addressed, lgtm

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/xen-chosen.dtsi (hunk 3 above)
Comment (otyshchenko1) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/xen-chosen.dtsi]: BTW, it seems that credit2 is used by default (CONFIG_SCHED_CREDIT2_DEFAULT=y), so "sched=credit2" property can be dropped also.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend (hunk 2 above)
Comment (aanisov) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend]: But it is the way XEN build system names xenpolicy file for rc builds.

Comment (aanisov) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend]: In xen/xsm/flask/Makefile you can find following:

POLICY_SRC := $(FLASK_BUILD_DIR)/xenpolicy-$(XEN_FULLVERSION)

Comment (aanisov) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xen/xen_git.bbappend]: So it might have unstable/rc suffixes.

**Comment by aanisov** [comment]
Pushed version employing `.inc` files for XEN version selection.

**Comment by aanisov** [comment]
Also relies on https://github.com/xen-troops/meta-xt-images/pull/112

**Pull request merged**